spatialprofilingtoolbox/db/feature_matrix_extractor.py

```diff
@@ -18,24 +18,61 @@
 
 class FeatureMatrixExtractor:
     """
-    Pull from the database and create convenience bundle of feature matrices
-    and metadata.
+    Pull from the database and create convenience bundle of feature matrices and metadata.
     """
-    @staticmethod
-    def extract(database_config_file, specimen: str=None, study: str=None, continuous_also=False):
-        E = FeatureMatrixExtractor
-        data_arrays = E.retrieve_expressions_from_database(database_config_file,
-                                                           specimen=specimen,
-                                                           study=study,
-                                                           continuous_also=continuous_also)
-        centroid_coordinates = E.retrieve_structure_centroids_from_database(database_config_file,
-                                                                            specimen=specimen,
-                                                                            study=study)
-        stratification = E.retrieve_derivative_stratification_from_database(database_config_file)
-        study_component_lookup = E.retrieve_study_component_lookup(database_config_file)
-        merged = E.merge_dictionaries(
-            E.create_feature_matrices(data_arrays, centroid_coordinates),
-            E.create_channel_information(data_arrays),
+
+    cursor: Psycopg2Cursor | None
+    database_config_file: str | None
+
+    def __init__(self,
+        cursor: Psycopg2Cursor | None=None,
+        database_config_file: str | None=None,
+    ):
+        if cursor is None and database_config_file is None:
+            logger.error('Must supply either cursor or database_config_file.')
+            return
+        if cursor is not None and database_config_file is not None:
+            message = 'A cursor and database configuration file were both specified. Using the '\
+                'cursor.'
+            logger.warning(message)
+        self.cursor = cursor
+        self.database_config_file = database_config_file
+
+    def extract(self,
+        specimen: str | None=None,
+        study: str | None=None,
+        continuous_also: bool=False,
+    ):
```

Comment:
Missing type hint for output.

Reply:
That's because the output "type" is very complicated. I don't think every final member is a dataframe, as you have supposed.

Reply:
This would of course benefit from refactoring to have clearer return types, but that's not really the point of this PR.